certora/specs/wadRayMath.spec

```
@@ -94,3 +96,31 @@ rule wadToRayLiveness(uint256 a) {
 
     assert lastReverted <=> a * WADTORAY() > 2^256;
 }
+
+/// wadWeightedAvg ///
+
+rule wadWeightedAvgSafety(uint256 x, uint256 y, uint256 w) {
+    uint res = wadWeightedAvg(x, y, w);
+
+    assert res == (x * (10^18 - w) + y * w + 10^18 / 2) / 10^18;
```

Not mandatory, but it is clean 
```suggestion
    assert res == (x * (WAD() - w) + y * w + WAD() / 2) / WAD();
```